### ISSUE TITLE
[BUG]  KerasSequenceValidater metrics initialization 

### ISSUE THREAD

**Issue by sararb** [NVIDIA/NVTabular#932]
**Describe the bug**
There is a mismatch between validation scores computed during training using KerasSequenceValidater callback and the scores returned by model.eval() after training.

**Steps/Code to reproduce bug**
This screenshot expose a training of LSTM model (Any other model can be used) with Training data built from Sigir eCom Challenge (Any data can be used). The red boxes show the difference of the AUC evaluation scores for the same validation set. 

<img width="1221" alt="BUG" src="https://user-images.githubusercontent.com/17721108/125004920-89ca9200-e028-11eb-8aae-9606a74f0d03.png">

**Expected behaviour**
The expected behaviour is that the scores returned by KerasSequenceValidater match exactly the ones returned by model.eval() after training is finished 

**Environment details (please complete the following information):**
 - Method of NVTabular install:  docker image  "nvcr.io/nvidian/prj-recsys/transformers4rec:0.2-hf-4.6.0-nvtabular-0.5.3"  with latest code of NVTabular pulled from main branch.
